lib/simpler/application.rb

```diff
@@ -22,18 +22,22 @@ def bootstrap!
       require_routes
     end
 
-    def routes(&block)
-      @router.instance_eval(&block)
-    end
-
     def call(env)
       route = @router.route_for(env)
+      return page_not_found unless route
+      
+      env['simpler.params'] ||= {}
+      env['simpler.params'].merge!(route.params(env))
```

Comment:
Ты забыл про параметры из форм и query string.
```suggestion
      request = Rack::Request.new(env)
      env['simpler.params'] = route.params(env).merge(request.params)
```